lib/src/git.rs

```diff
@@ -1226,9 +1226,8 @@ pub struct GitBranchPushTargets {
 
 pub struct GitRefUpdate {
     pub qualified_name: String,
-    // TODO: We want this to be a `current_target: Option<CommitId>` for the expected current
-    // commit on the remote. It's a blunt "force" option instead until git2-rs supports the
-    // "push negotiation" callback (https://github.com/rust-lang/git2-rs/issues/733).
+    pub expected_current_target: Option<CommitId>,
```

Comment:
nit: can you add doc comment that says `None` doesn't mean "no expectation", but "expects absent". That will probably help when we have to add real force push (i.e. `expected: Any`.)